pkg/apis/core/pods/helpers.go

```diff
@@ -19,9 +19,27 @@ package pods
 import (
 	"fmt"
 
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	api "k8s.io/kubernetes/pkg/apis/core"
 	"k8s.io/kubernetes/pkg/fieldpath"
 )
 
+// ContainerVisitor is called with each container spec, and returns true if visiting should continue.
+type ContainerVisitor func(container *api.Container, path *field.Path)
+
+// VisitContainers invokes the visitor function with a pointer to the container
+// spec of every container in the pod spec, include ephemeral containers.
+func VisitContainers(podSpec *api.PodSpec, visitor ContainerVisitor) {
```

Comment:
Same name, different signature, super confusing.  This should really be called `VisitContainersWithPath`